### PR TITLE
GHRSST module, and changes to netcdf-out to support pluggability and reduce code duplication

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                    CoverageView.CompositionType.BAND_SELECT);
+            coverageBands.add(band);
+        }
+        final CoverageView coverageView = new CoverageView(SST.getLocalPart(), coverageBands);
+        return coverageView;
+    }
+
+    /**
+     * Test NetCDF output from a NetCDF file with a rotated pole projection.
+     */
```
Comment (dromagnoli) [inline src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java]: I would change the comment for this test to reflect a GHRSST write instead.

File: src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            double EPS = 1e-3;
+            assertNotNull(globalAttributes.get("northernmost_latitude"));
+            assertNotNull(globalAttributes.get("southernmost_latitude"));
+            assertNotNull(globalAttributes.get("westernmost_longitude"));
+            assertNotNull(globalAttributes.get("easternmost_longitude"));
+            assertEquals(119.925, globalAttributes.get("northernmost_latitude").getNumericValue().doubleValue(), EPS);
+            assertEquals(-119.925, globalAttributes.get("southernmost_latitude").getNumericValue().doubleValue(), EPS);
+            assertEquals(-269.925, globalAttributes.get("westernmost_longitude").getNumericValue().doubleValue(), EPS);
+            assertEquals(269.925, globalAttributes.get("easternmost_longitude").getNumericValue().doubleValue(), EPS);
+            // resolution, take 2
```
Comment (dromagnoli) [inline src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java]: Are these lat and lon correct? lat +/-119 ?

File: src/extension/netcdf-out/src/main/java/org/geoserver/wcs/responses/NetCDFEncoder.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import org.geoserver.web.netcdf.NetCDFSettingsContainer;
+import org.geoserver.web.netcdf.layer.NetCDFLayerSettingsContainer;
+import ucar.ma2.InvalidRangeException;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Writes out a NetCDF file (the write parameters are provided during constructino, see {@link NetCDFEncoderFactory}
```
Comment (dromagnoli) [inline src/extension/netcdf-out/src/main/java/org/geoserver/wcs/responses/NetCDFEncoder.java]: small typo:
constructino -> construction

File: src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java (hunk 2 above)
Comment (aaime) [reply src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java]: As the comment says, it really has those bounds... it's a reduced version of a larger netcdf, compression the resolution so much I guess affected the bounds. But it's a match for gdalinfo output

File: src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java (hunk 1 above)
Comment (aaime) [reply src/community/netcdf-ghrsst/src/test/java/org/geoserver/wcs/responses/GHRSSTWCSTest.java]: Yep

**Pull request merged**